controllers/managedocs_controller.go

```diff
@@ -1067,6 +1085,52 @@ func (r *ManagedOCSReconciler) reconcilePrometheus() error {
 	return nil
 }
 
+func (r *ManagedOCSReconciler) getPrometheusRemoteWriteSpec() (*promv1.RemoteWriteSpec, error) {
+	if err := r.get(r.rhobsSecret); err != nil {
+		return nil, fmt.Errorf("unable to fetch rhobs secret: %v", err)
+	}
+	rhobsSecretData := r.rhobsSecret.Data
+	rhobsUserID := rhobsSecretData[rhobsClientId]
+	if rhobsUserID == nil {
+		return nil, fmt.Errorf("rhobs secret does not contain a %v entry", rhobsClientId)
+	}
+	rhobsUserAccessKey := rhobsSecretData[rhobsClientSecret]
+	if rhobsUserAccessKey == nil {
+		return nil, fmt.Errorf("rhobs secret does not contain a %v entry", rhobsClientSecret)
+	}
+
+	rhobsAudience := rhobsSecretData["rhobs-audience"]
+	if rhobsAudience == nil {
+		return nil, fmt.Errorf("rhobs secret does not contain a %v entry", "rhobs-audience")
+	}
+
+	remoteWriteSpec := &promv1.RemoteWriteSpec{
+		URL: r.RhobsApiUrl,
+		OAuth2: &promv1.OAuth2{
+			ClientSecret: corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: r.RhobsSecretName,
+				},
+				Key: rhobsClientSecret,
+			},
+			ClientID: promv1.SecretOrConfigMap{
+				Secret: &corev1.SecretKeySelector{
+					LocalObjectReference: corev1.LocalObjectReference{
+						Name: r.RhobsSecretName,
+					},
+					Key: rhobsClientId,
+				},
+			},
+			TokenURL: r.RhSSOTokenUrl,
+			EndpointParams: map[string]string{
+				"audience": string(rhobsAudience),
+			},
+		},
+		WriteRelabelConfigs: templates.PrometheusRemoteWriteRelabelTemplate,
+	}
```

Comment:
There are a lot of static parts here and only 6 really assignments of dynamic parts. 
Can we move all of the static parts into the template and just update the spec? 

This, of course, also means the function need to be split into two parts: 
1. Validation (the checks above) 
2. changing the spec - to be embedded in the prom reconcile func and have the following code
```
remoteWriteSpec := &r.prometheus.Spec.RemoteWrite
remoteWriteSpec.URL = r.RhobsApiUrl
remoteWriteSpec.OAuth2.ClientSecret.LocalObjectReference.Name := r.RhobsSecretName
remoteWriteSpec.OAuth2.ClientID.Secret.LocalObjectReference.Name := r.RhobsSecretName
remoteWriteSpec.OAuth2.TokenURL =  r.RhSSOTokenUrl
remoteWriteSpec.OAuth2.EndpointParams["audience"] = string(rhobsAudience)
```

Reply:
Done

Reply:
Done